System/Directory/Internal/Posix.hsc

```diff
@@ -49,8 +49,17 @@ c_PATH_MAX | c_PATH_MAX' > toInteger maxValue = Nothing
 c_PATH_MAX = Nothing
 #endif
 
+#if !defined(HAVE_REALPATH)
+
+c_realpath :: CString -> CString -> IO CString
+c_realpath _ _ = pure nullPtr
```

Comment:
If you return `nullPtr` you are supposed to also set `errno`. I'm not sure that's easily done in Haskell.

It'd be easier to just throw `UnsupportedOperation` directly into the `IO`.

Reply:
Fair, thanks for pointing out :)